node/src/types/block.rs

```diff
@@ -1111,6 +1113,209 @@ impl Item for BlockHeaderWithMetadata {
         self.block_header.height()
     }
 }
+#[derive(DataSize, Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq, Hash)]
+/// ID idenfitifying a request for a batch of block headers.
+pub struct BlockHeadersBatchId {
+    pub highest: u64,
```

Comment:
I think the block header batches should be hardcoded, with the height either being fixed or determined by the chainspec. For example, if that number is 1024, batch `0` is always going to be headers `genesis..1023`, batch `1` blocks `1024-2047` and so on. I think that would obviate a lot of checking.